packages/cli/src/commands/test-v2.ts

```diff
@@ -1,15 +1,69 @@
 import type { CommandModule } from "yargs";
-import { deployHandler, DeployOptions } from "./deploy-v2.js";
-import { yDeployOptions } from "./deploy-v2.js";
-import { anvil, forge, getRpcUrl, getTestDirectory } from "../utils/foundry.js";
+import readline from "readline";
+import chokidar from "chokidar";
+import { resolveConfigPath } from "@latticexyz/config";
 import chalk from "chalk";
 import { rmSync, writeFileSync } from "fs";
-import path from "path";
+import EventEmitter from "events";
+
+import { deployHandler, DeployOptions } from "./deploy-v2.js";
+import { yDeployOptions } from "./deploy-v2.js";
+import {
+  anvil,
+  forge,
+  FOUNDRY_OPTIONS_IGNORE_CODES,
+  getRpcUrl,
+  getSrcDirectory,
+  getTestDirectory,
+} from "../utils/foundry.js";
+import { CommandFailedError } from "../utils/errors.js";
+import { execLog } from "../utils/execLog.js";
 
-type Options = DeployOptions & { port?: number; worldAddress?: string; forgeOptions?: string };
+type Options = DeployOptions & {
+  port?: number;
+  watch?: boolean;
+  forgeOptions?: string;
+};
 
 const WORLD_ADDRESS_FILE = ".mudtest";
 
+enum Events {
+  TESTS_CHANGED = "TESTS_CHANGED",
+  SOURCES_CHANGED = "SOURCES_CHANGED",
+  CONFIG_CHANGED = "CONFIG_CHANGED",
+}
+
+const listenForKeyPressesOrEvents = (keys: string[], exit: string, events: EventEmitter) => {
+  return new Promise<{ keyPressed?: string; filesChanged: string[] }>((resolve, reject) => {
+    const removeListenersAndResolve = (returnValue: { keyPressed?: string; filesChanged: string[] }) => {
+      events.removeAllListeners();
+      resolve(returnValue);
+    };
+
+    events.on(Events.TESTS_CHANGED, (filesChanged) => removeListenersAndResolve({ filesChanged }));
+    events.on(Events.SOURCES_CHANGED, (filesChanged) => removeListenersAndResolve({ filesChanged }));
+    events.on(Events.CONFIG_CHANGED, (filesChanged) => removeListenersAndResolve({ filesChanged }));
```

Comment:
I know we're removing these immediately, but I tend to use `.once` for cases like these just to show intent and to reduce the chance of memory leaks if this behavior changes in the future